src/main/java/org/craftercms/studio/impl/v2/job/StudioSyncRepositoryTask.java

```diff
@@ -87,14 +87,15 @@ private void syncRepository(String site) throws SiteNotFoundException, IOExcepti
                 if (Objects.nonNull(gl)) {
                     String lastRepoCommitId = contentRepository.getRepoLastCommitId(site);
                     if (StringUtils.equals(lastRepoCommitId, lastProcessedCommit)) {
-                        contentRepository.markGitLogVerifiedProcessed(site, gl.getCommitId());
+                        if (gl.getProcessed() == 0) {
+                            contentRepository.markGitLogVerifiedProcessed(site, gl.getCommitId());
+                        }
                         contentRepository.markGitLogProcessedBeforeMarker(site, gl.getId(), 1);
```

Comment:
Is there a reason why `contentRepository.markGitLogProcessedBeforeMarker(site, gl.getId(), 1);` is not inside the `if`? This SQL statement seems to be the heaviest from the too.

Reply:
that one marks all unprocessed commit ids